src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts

```diff
@@ -33,72 +41,211 @@ export class HuiDialogEditCard extends LitElement {
-    this._save = this._save.bind(this);
-  }
+  @property() private _saving: boolean = false;
 
   public async showDialog(params: EditCardDialogParams): Promise<void> {
     this._params = params;
     const [view, card] = params.path;
-    this._newCard = card !== undefined ? false : true;
     this._cardConfig =
       card !== undefined
         ? params.lovelace.config.views[view].cards![card]
         : undefined;
   }
 
+  private get _cardEditorEl(): HuiCardEditor | null {
+    return this.shadowRoot!.querySelector("hui-card-editor");
+  }
+
   protected render(): TemplateResult | void {
     if (!this._params) {
       return html``;
     }
-    if (!this._cardConfig) {
-      // Card picker
-      return html`
-        <hui-dialog-pick-card
-          .hass="${this.hass}"
-          .cardPicked="${this._cardPicked}"
-          .closeDialog="${this._cancel}"
-        ></hui-dialog-pick-card>
-      `;
-    }
+
     return html`
-      <hui-edit-card
-        .hass="${this.hass}"
-        .lovelace="${this._params.lovelace}"
-        .cardConfig="${this._cardConfig}"
-        .closeDialog="${this._cancel}"
-        .saveCard="${this._save}"
-        .newCard="${this._newCard}"
-      >
-      </hui-edit-card>
+      <ha-paper-dialog with-backdrop opened modal>
+        <h2>
+          ${this.hass!.localize("ui.panel.lovelace.editor.edit_card.header")}
+        </h2>
+        <paper-dialog-scrollable>
+          ${this._cardConfig === undefined
+            ? html`
+                <hui-card-picker
+                  .hass="${this.hass}"
+                  @config-changed="${this._handleConfigChanged}"
+                ></hui-card-picker>
+              `
+            : html`
+                <div class="content">
+                  <div class="element-editor">
+                    <hui-card-editor
+                      .hass="${this.hass}"
+                      .value="${this._cardConfig}"
+                      @config-changed="${this._handleConfigChanged}"
+                    ></hui-card-editor>
+                  </div>
+                  <hui-card-preview
+                    .hass="${this.hass}"
+                    .config="${this._cardConfig}"
+                  ></hui-card-preview>
+                </div>
+              `}
+        </paper-dialog-scrollable>
+        <div class="paper-dialog-buttons">
+          <mwc-button @click="${this._close}">
+            ${this.hass!.localize("ui.common.cancel")}
+          </mwc-button>
+          <mwc-button ?disabled="${!this._canSave}" @click="${this._save}">
+            <paper-spinner
+              ?active="${this._saving}"
+              alt="Saving"
+            ></paper-spinner>
+            ${this.hass!.localize("ui.common.save")}
+          </mwc-button>
+        </div>
+      </ha-paper-dialog>
     `;
   }
 
-  private _cardPicked(cardConf: LovelaceCardConfig): void {
-    this._cardConfig = cardConf;
+  static get styles(): CSSResult[] {
```

Comment:
Use `CSSResultArray`, the result is actually flattened so the `CSSResultArray` type allows that.